src/utils/url.ts

```diff
@@ -0,0 +1,42 @@
+import { DwnError, DwnErrorCode } from '../core/dwn-error.js';
+
+const URL_PROTOCOL_REGEX = /^[^:]+:\/\/./;
```

Comment:
For my understanding: why this particular regex? Is this something you created or is it from some well-known regex for URIs? It just looks too lenient. e.g. this string `??://asdf.` would pass as valid...

Reply:
Oh, I see. you are only checking the "scheme" portion later in code. I thought "protocol" here meant DWN protocol for some reason. I'd suggest to encapsulate the regex within `normalizeProtocolUri(...)` as reusability is probably unlikely.